homeassistant/components/sharkiq/update_coordinator.py

```diff
@@ -0,0 +1,93 @@
+"""Data update coordinator for shark iq vacuums."""
+
+from typing import Dict, List
+
+from async_timeout import timeout
+from sharkiqpy import (
+    AylaApi,
+    SharkIqAuthError,
+    SharkIqAuthExpiringError,
+    SharkIqNotAuthedError,
+    SharkIqVacuum,
+)
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import API_TIMEOUT, DOMAIN, LOGGER, UPDATE_INTERVAL
+
+
+class SharkIqUpdateCoordinator(DataUpdateCoordinator):
+    """Define a wrapper class to update Shark IQ data."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        config_entry,
+        ayla_api: AylaApi,
+        shark_vacs: List[SharkIqVacuum],
+    ) -> None:
+        """Set up the SharkIqUpdateCoordinator class."""
+        self.ayla_api = ayla_api
+        self.shark_vacs = {
+            sharkiq.serial_number: sharkiq for sharkiq in shark_vacs
+        }  # type: Dict[SharkIqVacuum]
+        self._config_entry = config_entry
+        self._online_dsns = {}
+
+        super().__init__(hass, LOGGER, name=DOMAIN, update_interval=UPDATE_INTERVAL)
+
+    def is_online(self, dsn: str) -> bool:
```

Comment:
One more to cover in your tests to 100% this class :smiley: 

Also, should this be a `@property`?

Reply:
I probably should add a property to cover `self._online_dsns`, but this method needs an argument since a single coordinator (corresponding to an API account) can have multiple devices.   I'll try to get to the tests this week.